src/main/java/de/tum/in/www1/artemis/domain/enumeration/SubmissionType.java

```diff
@@ -1,7 +1,9 @@
 package de.tum.in.www1.artemis.domain.enumeration;
 
 /**
- * The SubmissionType enumeration.
+ * MANUAL: submission triggered by a student
+ * EXTERNAL: submission that was added through the add external submission dialogue
+ * ILLEGAL: submission of an exam programming exercise that is submitted after the individual exercise end date
```

Comment:
Maybe add "or before the individual exercise start" to the javadoc for ILLEGAL :)

Reply:
Thanks for the heads up 👍

Reply:
Fixed in https://github.com/ls1intum/Artemis/pull/3701/commits/ae4471168560bd6d935f9b3035361869022145b2